qa/tasks/cephfs/xfstests_dev.py

```diff
@@ -1,31 +1,60 @@
 from io import StringIO
+from os.path import join
 from logging import getLogger
 
 from tasks.cephfs.cephfs_test_case import CephFSTestCase
 
 
-logger = getLogger(__name__)
+log = getLogger(__name__)
 
 
-# TODO: add code to run non-ACL tests too.
-# TODO: get tests running with SCRATCH_DEV and SCRATCH_DIR.
 # TODO: make xfstests-dev tests running without running `make install`.
 # TODO: make xfstests-dev compatible with ceph-fuse. xfstests-dev remounts
 # CephFS before running tests using kernel, so ceph-fuse mounts are never
 # actually testsed.
 class XFSTestsDev(CephFSTestCase):
 
+    #
+    # Following are the methods that download xfstests-dev repo and get it
+    # ready to run tests from it.
+    #
+
     def setUp(self):
         super(XFSTestsDev, self).setUp()
-        self.prepare_xfstests_dev()
-
-    def prepare_xfstests_dev(self):
         self.get_repo()
         self.get_test_and_scratch_dirs_ready()
         self.install_deps()
         self.create_reqd_users()
         self.write_local_config()
+        self.build_and_install()
 
+    def tearDown(self):
+        self.del_users_and_groups()
+        self.save_results_dir()
+        self.mount_a.client_remote.run(args=f'sudo rm -rf {self.repo_path}',
+                                       omit_sudo=False, check_status=False)
+        self.save_results_dir()
+        super(XFSTestsDev, self).tearDown()
+
+    def del_users_and_groups(self):
+        self.mount_a.client_remote.run(args=['sudo', 'userdel', '--force',
+                                             '--remove', 'fsgqa'],
+                                       omit_sudo=False, check_status=False)
+        self.mount_a.client_remote.run(args=['sudo', 'userdel', '--force',
+                                             '--remove', '123456-fsgqa'],
+                                       omit_sudo=False, check_status=False)
+        self.mount_a.client_remote.run(args=['sudo', 'groupdel', 'fsgqa'],
+                                       omit_sudo=False, check_status=False)
+
+    def save_results_dir(self):
+        src = join(self.repo_path, 'results')
+        dst = join('/var/log/ceph', 'xfstests-dev-results')
+        if self.mount_a.run_shell(f'sudo stat {src}',
```

Comment:
log if for whatever reason `results` is absent.

Reply:
Umm... Should we? Because the only cause for results dir will be absent is when tests are not run.

Reply:
The verbosity might make debugging easier though.